e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+            FilePayload,
+            typing.List[typing.Union[str, pathlib.Path]],
+            typing.List[FilePayload],
+        ],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self.loc.set_input_files(file_path, timeout=timeout)
+
+    def expect_files(
+        self,
+        files: typing.Union[str, typing.List[str], None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        # TODO-barret; Find method to get files, or remove method
+        # TODO-barret; Test value being sent to shiny?
+        NotImplementedError("`expect_files()` is not implemented")
+
+    def expect_multiple(self, multiple: bool, *, timeout: Timeout = None) -> None:
+        expect_multiple(self.loc, multiple, timeout=timeout)
+
+    def expect_accept(
+        self,
+        accept: typing.Union[typing.List[str], AttrValue],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if isinstance(accept, typing.List):
+            accept = ",".join(accept)
+        expect_attr(self.loc, "accept", accept, timeout=timeout)
+
+    def expect_width(self, width: StyleValue, *, timeout: Timeout = None) -> None:
+        expect_el_style(self.loc_container, "width", width, timeout=timeout)
+
+    def expect_button_label(
+        self,
+        button_label: TextValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if button_label is None:
+            button_label = ""
+        playwright_expect(self.loc_button).to_have_text(button_label, timeout=timeout)
+
+    def expect_capture(
+        self,
+        capture: typing.Union[Literal["environment", "user"], None],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc, "capture", capture, timeout=timeout)
+
+    def expect_placeholder_to_have_value(
+        self, value: AttrValue, *, timeout: Timeout = None
+    ) -> None:
+        expect_attr(self.loc_file_display, "placeholder", value=value, timeout=timeout)
+
+
+class InputSlider(_WidthLoc, _InputWithLabel):
+    # id: str,
+    # label: TagChildArg,
+    # min: SliderValueArg,
+    # max: SliderValueArg,
+    # value: Union[SliderValueArg, Iterable[SliderValueArg]],
+    # step: Optional[SliderStepArg] = None,
+    # ticks: bool = True,
+    # animate: Union[bool, AnimationOptions] = False,
+    # width: Optional[str] = None,
+    # sep: str = ",",
+    # pre: Optional[str] = None,
+    # post: Optional[str] = None,
+    # time_format: Optional[str] = None,
+    # timezone: Optional[str] = None,
+    # drag_range: bool = True,
+
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"input#{id}",
+            loc_label=f"label#{id}-label",
+        )
+
+    def expect_value(self, value: AttrValue, *, timeout: Timeout = None) -> None:
+        # TODO-barret; implement
+        NotImplementedError("Need to get the value somehow")
```

Comment:
This could just be the handle text to find the value; if we consider this to be just "what does the user see with their eyeballs". Also note that range sliders have two handles.

Reply:
a739fe3

Reply:
Resolving this comment as we'll need to make a `InputSliderRange` class which should have two values. 
Implemented: `InputSlider.expect_value(value: PatternOrStr)` using the label above the slider handle